walkers/src/download.rs

```diff
@@ -72,19 +108,62 @@ where
     // Keep outside the loop to reuse it as much as possible.
     let client = http_client(http_options);
 
+    let mut ongoing_downloads = Vec::<_>::new();
+
     loop {
-        let request = request_rx.next().await.ok_or(())?;
-        let url = source.tile_url(request);
+        let request = request_rx.next();
+        let below_concurrent_downloads_limit = ongoing_downloads.len() < 2;
```

Comment:
I am not sure that this is correct the right approach: here we are trying to limit the number of downloads, not connections.
in particular, this related to http/2 multiplexing